nucypher/blockchain/eth/networks.py

```diff
@@ -1,57 +1,83 @@
-class NetworksInventory:  # TODO: See #1564
-
-    MAINNET = "mainnet"
-    LYNX = "lynx"
-    IBEX = "ibex"  # this is required for configuration file migrations (backwards compatibility)
-    ETH = "ethereum"
-    TAPIR = "tapir"
-    ORYX = "oryx"
-
-    # TODO: Use naming scheme to preserve multiple compatibility with multiple deployments to a single network?
-    POLYGON = 'polygon'
-    MUMBAI = 'mumbai'
-
-    UNKNOWN = 'unknown'  # TODO: Is there a better way to signal an unknown network?
-    DEFAULT = MAINNET
-
-    __to_chain_id_eth = {
-        MAINNET: 1,  # Ethereum Mainnet
-        IBEX: 5,  # this is required for configuration file migrations (backwards compatibility)
-        LYNX: 5,  # Goerli
-        TAPIR: 5,
-        ORYX: 5
-    }
-    __to_chain_id_polygon = {
-        # TODO: Use naming scheme?
-        POLYGON: 137,    # Polygon Mainnet
-        MUMBAI: 80001,   # Polygon Testnet (Mumbai)
-    }
-
-    ETH_NETWORKS = tuple(__to_chain_id_eth.keys())
-    POLY_NETWORKS = tuple(__to_chain_id_polygon.keys())
-
-    NETWORKS = ETH_NETWORKS + POLY_NETWORKS
-
-    class UnrecognizedNetwork(RuntimeError):
-        pass
+from enum import Enum
+from typing import List, NamedTuple
 
-    @classmethod
-    def get_ethereum_chain_id(cls, network):
-        try:
-            return cls.__to_chain_id_eth[network]
-        except KeyError:
-            raise cls.UnrecognizedNetwork(network)
+from nucypher.config.constants import TEMPORARY_DOMAIN
+
+
+class ChainInfo(NamedTuple):
+    chain_id: int
+    chain_name: str
```

Comment:
Since you're already in a "chain" namespace here, you don't need to smurf.  Consider `chain_id` -> `id` and `chain_name` -> `name`.  In implementation this will be:

```python
chain_info = ChainInfo(name='lynx', id=5)
_id, name = chain_info.id, chain_info.name
```

Reply:
✔️